src/client.cpp

```diff
@@ -63,6 +63,18 @@ void set_client_timer(struct itimerval *timer) {
     timer->it_interval.tv_usec = 0;
 }
 
+//------------------------------------------------------------------------------
+/* set the timer on client based on [-o observation] parameter given by user */
+void set_observation_timer(struct itimerval *timer) {
+    uint32_t total_observations = s_user_params.warmup_obs + s_user_params.cooldown_obs +
+                                  g_pApp->m_const_params.observation_test_count;
+    uint32_t waiting_cap = (total_observations >> 12) + 1; // Gurantee at least 1 sec
```

Comment:
The function for calculating `waiting_cap` should change. Per discussion, the IP transport layer type and message size should be accounted for here. Reviewers, please provide feedback on calculating `waiting_cap`.